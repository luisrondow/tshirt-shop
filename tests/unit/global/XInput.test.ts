import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'

import { XInput } from '#components'

describe('XInput component', () => {
  it('should render correctly', () => {
    const wrapper = mount(XInput)

    expect(wrapper.vm).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render correctly with "value"', () => {
    const wrapper = mount(XInput, {
      props: {
        modelValue: 'test',
      },
    })

    expect(wrapper.find('input').element.value).toBe('test')
  })

  it('should render correctly with "placeholder" props', () => {
    const wrapper = mount(XInput, {
      props: {
        placeholder: 'test',
      },
    })

    expect(wrapper.find('input').attributes('placeholder')).toBe('test')
  })

  it('should emit "update:modelValue" event', async () => {
    const wrapper = mount(XInput)

    await wrapper.find('input').setValue('test')

    expect(wrapper.find('input').element.value).toBe('test')
    expect(wrapper.emitted('update:modelValue')).toBeTruthy()
    expect(wrapper.emitted('update:modelValue')?.[0]).toEqual(['test'])
  })
})
