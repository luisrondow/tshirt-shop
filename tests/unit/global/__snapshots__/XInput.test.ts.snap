// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`XInput component > should render correctly 1`] = `
<input
  class="h-12 w-full border border-gray-400 bg-white px-4 text-black outline-none placeholder:text-gray-300 focus:border-black"
  value=""
/>
`;
